### ISSUE TITLE
cross-feeding and which medium should I use?

### ISSUE THREAD

**Issue by mhjonathan** [jotech/gapseq#161]
Before my questions, I would like to thank to all developers of this wonderful tool :)

I'm trying to check whether there is positive/negative correlation in growth  between Species A and Species B.

After having draft model of each species, I'm stuck at which medium should I use for gapfilling. What I'm thinking of is to merge respectively predicted medium (medium for A and medium for B), and use this merged media for gapfilling.

But I'm not sure how to merge the predicted medium. Is it just like, "cat speciesA-medium.csv speciesB-medium.csv > merged-medium.csv"? 

Thank you in advance!

**Comment by jotech** [comment]
Hi @mhjonathan 
Thank you for using gapseq :)
I would argue that since microbial interactions are often context-dependent, defining the conditions under which you are looking for growth influences makes a lot of sense. 

Do you have an actual experiment with these two species in mind? 
Then go for the nutritional context of the experiment (growth medium). Otherwise, a simple merge of the predicted growth models might be a good way to start! (but please be aware of duplicates if you do it with `cat`)

**Comment by mhjonathan** [comment]
Wonderful. What I hypothesised is that Species A produces some kinda metabolic product which could affect the growth of Species B. As the data is originated from human gut, I want to think of 1) normal conditions of human gut environment 2) nutrient rich environment to control potential nutrient-deprived condition for checking species's own effect. 

Would be my second thought matched with the case of merged predicted medium?

**Comment by jotech** [comment]
I agree, merging the growth media might work in your case! If you expect the interaction to be syntrophic (i.e., the product of A is strictly necessary for B's growth), then you should add the product of A into the medium, too.

For the conditions of the human gut, we also have a predefined medium:
https://github.com/jotech/gapseq/blob/master/dat/media/gut.csv

**Comment by mhjonathan** [comment]
Thank you for sharing the predefined medium! That really helps :)

And one more question: 
You seem to work on BacArena also, so if I want to do a community analysis like Species A, Species B, Species C at one Arena and wanna see their growth difference under optimal condition, would it be fine to write the code below?:

```
arena_com <- addOrg(arena_com, Species_A_model, amount=20)
arena_com <- addOrg(arena_com, Species_B_model, amount=20)
arena_com <- addOrg(arena_com, Species_C_model, amount=20)
  
arena_com <- addDefaultMed(arena_com, Species_A_model, unit='fmol/cell')
arena_com <- addDefaultMed(arena_com, Species_B_model, unit='fmol/cell')
arena_com <- addDefaultMed(arena_com, Species_C_model, unit='fmol/cell')

arena_com <- addSubs(arena_com, smax = arena_subs$maxFlux*3, mediac=arena_subs$ex.rxn, unit="mM", add=T )
```

What I wonder is whether I should use addSubs command as I think addDefaultMed already adds merged medium (predicted by gapseq) components with addDefaultMed command.

**Comment by jotech** [comment]
I would use the medium already for gap-filling to ensure that all organisms can grow under the medium conditions.

In BacArena, `addDefaultMed()` could work, but it is not additive, so concentrations are potentially overwritten if you use it for different organisms! I would recommend employing `addSubs()` for this use case.

**Comment by mhjonathan** [comment]
I get your point, but what I also want to see is that whether their metabolism is changed under gut-like environment (non-optimal) cuz there are some species secreting toxins/different substrates under stressed/nutrient-deprived status. Wouldn't it be found with _in silico_ simulation?

> In BacArena, addDefaultMed() could work, but it is not additive, so concentrations are potentially overwritten if you use it for different organisms! I would recommend employing addSubs() for this use case.
Thank you for the answer!

**Comment by jotech** [comment]
To your first question, I would not expect to see any toxin or stress effects with metabolic models because the modeling approach does not cover it. But I'm not completely sure if I got your question correctly?

To your second question. Yes, the gut medium contains a broader range of substrates, while the predicted medium is more essential. For example if amino acids can be synthesized then the predicted growth medium does not contain them and, thus, they must be produced which costs energy and material and causes the growth rate to be smaller.

**Comment by mhjonathan** [comment]
> To your first question, I would not expect to see any toxin or stress effects with metabolic models because the modeling approach does not cover it. But I'm not completely sure if I got your question correctly?

What I meant is their substrate preferences under optimal/deprived environment and final product following the preferences. 

And about the gut-like predefined medium you provided, can you tell me its source? If it is from your own experiment data that unavailable to be opened, it's absolutely fine :)
